### PR TITLE
Export spans to hec

### PR DESCRIPTION
**Pull request by atoulme**
Description:
Adds trace exports support to Splunk HEC.

Testing:
Unit tests.

Documentation:
No doc - this is complementary to the metric export support with no additional config keys.

### CODE REVIEW

**Comment by atoulme** [comment]
Opened after closing #323 to trigger coverage.

**Comment by rmfitzpatrick** [comment]
@atoulme, thanks for your patience.  Would you mind rebasing to generate new hashes?  I believe that will engage codecov (tested in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/360).

File: exporter/splunkhecexporter/exporter.go [hunk 1]
```diff
@@ -50,12 +59,11 @@ type exporterOptions struct {
 	token string
 }
 
-// New returns a new Splunk exporter.
-func New(
+// new returns a new Splunk exporter.
```
Comment (rmfitzpatrick) [inline exporter/splunkhecexporter/exporter.go:62]: nit: stale name in description

Comment (atoulme) [reply exporter/splunkhecexporter/exporter.go]: Fixed, thanks :)

**Review by rmfitzpatrick — APPROVED** [review]

File: exporter/splunkhecexporter/tracedata_to_splunk.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		host = data.Resource.Labels[hostnameLabel]
+	}
+	if host == "" {
+		host = unknownHostName
+	}
+	numDroppedSpans := 0
+	splunkEvents := make([]*splunkEvent, 0)
+	for _, span := range data.Spans {
+		if span.StartTime == nil {
+			logger.Warn(
```
Comment (tigrannajaryan) [inline exporter/splunkhecexporter/tracedata_to_splunk.go:43]: This is not recommended: https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md#logging

`logger.Debug` may be a better choice.

File: exporter/splunkhecexporter/tracedata_to_splunk.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				"Span dropped as it had no start timestamp",
+				zap.Any("span", span))
+			numDroppedSpans++
+			continue
+		}
+		se := &splunkEvent{
+			Time:       timestampToEpochMilliseconds(span.StartTime),
```
Comment (tigrannajaryan) [inline exporter/splunkhecexporter/tracedata_to_splunk.go:50]: I may be reading HEC spec incorrectly but this does not look right. AFAIK millisecond precision should use decimal point representation in seconds, not integer milliseconds: https://docs.splunk.com/Documentation/Splunk/8.0.4/Data/FormateventsforHTTPEventCollector

File: exporter/splunkhecexporter/client.go [hunk 4]
```diff
@@ -94,6 +97,70 @@ func (c *client) pushMetricsData(
...
+	if len(splunkEvents) == 0 {
+		return numDroppedSpans, nil
+	}
+
+	body, compressed, err := encodeBodyEvents(&c.zippers, splunkEvents, c.config.DisableCompression)
+	if err != nil {
+		return len(td.Spans), consumererror.Permanent(err)
+	}
+
+	// TODO the client sends synchronously data as of now. It would make more sense to buffer data coming in, and send batches as supported by Splunk. To batch effectively, a ring buffer approach would work well - based on time and number of messages queued.
```
Comment (tigrannajaryan) [inline exporter/splunkhecexporter/client.go:117]: This should not be necessary once the exporter is upgraded to use the new trace data type. The new type is capable of representing batches from multiple resources and coupled with recommended `batch` processor should no longer require any batching here.

File: exporter/splunkhecexporter/tracedata_to_splunk.go (hunk 2 above)
Comment (atoulme) [reply exporter/splunkhecexporter/tracedata_to_splunk.go]: Changed, thanks.

File: exporter/splunkhecexporter/tracedata_to_splunk.go (hunk 3 above)
Comment (atoulme) [reply exporter/splunkhecexporter/tracedata_to_splunk.go:50]: Thanks for catching this. I have updated the code and added a couple tests to make sure the conversion works well.

File: exporter/splunkhecexporter/client.go (hunk 4 above)
Comment (atoulme) [reply exporter/splunkhecexporter/client.go]: Removed the TODOs for now.

File: exporter/splunkhecexporter/tracedata_to_splunk.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+package splunkhecexporter
+
+import (
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.uber.org/zap"
+)
+
+type splunkEvent struct {
+	Time       int64       `json:"time"`                 // epoch time
```
[the remark this answers is not in the document]
Comment (atoulme) [reply exporter/splunkhecexporter/tracedata_to_splunk.go]: Fixed, moved to a float64, and make sure time computation looks correct.

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Pull request merged**